src/Main.hs

```diff
@@ -108,55 +107,58 @@ main = do
 
-            results <- flip mapM (queries claims)
+            results <- forM (queries claims)
                           (\(i, qs) -> do
-                            rs <- mapM (satWithTimeOut solver smttimeout debug) qs
+                            rs <- mapM (satWithTimeOut solver' smttimeout' debug') qs
                             pure (i, rs)
                           )
             mapM_ handleResults results
 
       (Coq f) -> do
         contents <- readFile f
         proceed "" (parse (lexer contents)) $ \untyped ->
-          proceed "" (typecheck untyped) $ \typed ->
+          proceed "" (enrich <$> (typecheck untyped)) $ \typed ->
             TIO.putStr $ coq (lookupVars untyped) typed
 
-      (K spec soljson gas storage extractbin out) -> do
-        specContents <- readFile spec
-        solContents  <- readFile soljson
-        let kOpts = KOptions (maybe mempty Map.fromList gas) storage extractbin
-            errKSpecs = do refinedSpecs  <- parse (lexer specContents) >>= typecheck
+      (K spec' soljson' gas' storage' extractbin' out') -> do
+        specContents <- readFile spec'
+        solContents  <- readFile soljson'
+        let kOpts = KOptions (maybe mempty Map.fromList gas') storage' extractbin'
+            errKSpecs = do refinedSpecs <- compile specContents
                            (sources, _, _) <- errMessage (nowhere, "Could not read sol.json")
                              $ Solidity.readJSON $ pack solContents
                            forM (catBehvs refinedSpecs)
                              $ makekSpec sources kOpts (catInvs refinedSpecs)
         proceed specContents errKSpecs $ \kSpecs -> do
-          let printFile (filename, content) = case out of
+          let printFile (filename, content) = case out' of
                 Nothing -> putStrLn (filename <> ".k") >> putStrLn content
                 Just dir -> writeFile (dir <> "/" <> filename <> ".k") content
           forM_ kSpecs printFile
 
 -- cvc4 sets timeout via a commandline option instead of smtlib `(set-option)`
 satWithTimeOut :: Maybe Text -> Maybe Integer -> Bool -> Symbolic () -> IO SatResult
-satWithTimeOut solver maybeTimeout debug' sym
-  | solver == Just "cvc4" = do
-      setEnv "SBV_CVC4_OPTIONS" ("--lang=smt --incremental --interactive --no-interactive-prompt --model-witness-value --tlimit-per=" <> show timeout)
-      res <- satWith cvc4{verbose=debug'} sym
-      setEnv "SBV_CVC4_OPTIONS" ""
-      return res
-  | solver == Just "z3" = runwithz3
-  | solver == Nothing = runwithz3
-  | otherwise = error "Unknown solver. Currently supported solvers; z3, cvc4"
- where timeout = fromMaybe 20000 maybeTimeout
-       runwithz3 = satWith z3{verbose=debug'} $ (setTimeOut timeout) >> sym
+satWithTimeOut solver' maybeTimeout debug' sym = case solver' of
+  Just "cvc4" -> do
+    setEnv "SBV_CVC4_OPTIONS" ("--lang=smt --no-interactive-prompt --model-witness-value --tlimit-per=" <> show timeout)
```

Comment:
hmm, why did you remove `--incremental` and `--interactive`/

Reply:
oh, I had thought they were only useful in the context of `hevm`, where we are using the `query` monad and `push`/`pop`?

So far in `act` we just submit queries to the solver fully formed and call `check-sat` once per query. Did I misunderstand something?

Reply:
We might be able to get away without them, but keeping them doesn't seem very problematic either. I put them all there because those were [sbvs default options](https://github.com/LeventErkok/sbv/blob/master/Data/SBV/Provers/CVC4.hs#L31)

Reply:
ah ok, I'll put them back.